source/merge.ts

```diff
@@ -41,38 +37,3 @@ export default function merge<Target extends Record<string, unknown>, Source ext
 
 	return target as Target & Source;
 }
-
-export function mergeOptions<T extends Options>(...sources: T[]): T & { hooks: Partial<Hooks> } {
-	const mergedOptions = merge({} as T & { hooks: Partial<Hooks> }, ...sources.map(source => source || {}));
-
-	const hooks = knownHookEvents.reduce((acc, current) => ({...acc, [current]: []}), {}) as Record<HookEvent, HookType[]>;
-
-	for (const source of sources) {
-		// We need to check `source` to allow calling `.extend()` with no arguments.
-		if (source && source.hooks) {
-			for (const hook of knownHookEvents) {
-				hooks[hook] = hooks[hook].concat(source.hooks[hook] || []);
-			}
-		}
-	}
-
-	mergedOptions.hooks = hooks as Hooks;
-
-	return mergedOptions;
-}
-
-export function mergeInstances(instances: Got[], methods?: Method[]): Defaults {
-	const handlers = instances.map(instance => instance.defaults.handler);
-	const size = instances.length - 1;
-
-	return {
-		methods,
-		options: mergeOptions(...instances.map(instance => instance.defaults.options || {})),
-		handler: <T extends ProxyStream | CancelableRequest<Response>>(options: NormalizedOptions, next: (options: NormalizedOptions) => T) => {
-			let iteration = 0;
-			const iterate = (newOptions: NormalizedOptions): T => handlers[++iteration]!(newOptions, iteration === size ? next : iterate);
-
-			return iterate(options);
-		}
-	};
-}
```

Comment:
We need to move this out because if we are going to pre-normalize all options before merging, we need to avoid `require` collisions (self-requiring files).